flocker/control/schema/endpoints.yml

```diff
@@ -12,3 +12,61 @@ definitions:
     required:
       - flocker
     additionalProperties: false
+
+  datasets:
+    type: object
+    properties:
+      dataset_id:
+        title: "Unique identifier"
+        description: |
+          An opaque identifier, unique across the cluster, identifying a
+          particular dataset.  If not given, a new identifier will be generated
+          and returned.
+        type: string
+        # The length of a stringified uuid
+        minLength: 36
+        maxLength: 36
+
+      metadata:
+        title: "Data about a dataset"
+        description: |
+          Additional key/value data describing the dataset.  These items are
+          not interpreted by Flocker.  If not given, no metadata will be
+          associated with the new dataset.
+        type: object
+        patternProperties:
+          ".{1,256}":
```

Comment:
There does appear to be a test for this, so I'm puzzled why it isn't failing.

Also, it appears that if you want to *disallow* properties that don't match, you also need `additionalProperties: false`. Otherwise, this only says that properties that do match this must match a schema, and says nothing about properties that don't.